helix-term/src/commands.rs

```diff
@@ -4240,3 +4329,246 @@ fn suspend(_cx: &mut Context) {
+        if let Some(pos) = breakpoints.iter().position(|b| b.line == breakpoint.line) {
+            breakpoints.remove(pos);
+        } else {
+            breakpoints.push(breakpoint);
+        }
+
+        let breakpoints = breakpoints.clone();
+
+        let request = debugger.set_breakpoints(path, breakpoints);
+        let _ = block_on(request).unwrap();
+    }
+}
+
+fn dap_run(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    if let Some(debugger) = &mut cx.editor.debugger {
+        if debugger.is_running {
+            cx.editor
+                .set_status("Debuggee is already running".to_owned());
+            return;
+        }
+        let request = debugger.configuration_done();
+        let _ = block_on(request).unwrap();
+        debugger.is_running = true;
+    }
+}
+
+fn dap_continue(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    if let Some(debugger) = &mut cx.editor.debugger {
+        if debugger.is_running {
+            cx.editor
+                .set_status("Debuggee is already running".to_owned());
+            return;
+        }
+
+        let request = debugger.continue_thread(debugger.stopped_thread.unwrap());
+        let _ = block_on(request).unwrap();
+        debugger.is_running = true;
+        debugger.stack_pointer = None;
+    }
+}
+
+fn dap_pause(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    if let Some(debugger) = &mut cx.editor.debugger {
+        if !debugger.is_running {
+            cx.editor.set_status("Debuggee is not running".to_owned());
+            return;
+        }
+
+        // FIXME: correct number here
+        let request = debugger.pause(0);
+        let _ = block_on(request).unwrap();
+    }
+}
+
+fn dap_in(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    if let Some(debugger) = &mut cx.editor.debugger {
+        if debugger.is_running {
+            cx.editor
+                .set_status("Debuggee is already running".to_owned());
+            return;
+        }
+
+        let request = debugger.step_in(debugger.stopped_thread.unwrap());
+        let _ = block_on(request).unwrap();
+    }
+}
+
+fn dap_out(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    if let Some(debugger) = &mut cx.editor.debugger {
+        if debugger.is_running {
+            cx.editor
+                .set_status("Debuggee is already running".to_owned());
+            return;
+        }
+
+        let request = debugger.step_out(debugger.stopped_thread.unwrap());
+        let _ = block_on(request).unwrap();
+    }
+}
+
+fn dap_next(cx: &mut Context) {
+    use helix_lsp::block_on;
+
+    if let Some(debugger) = &mut cx.editor.debugger {
+        if debugger.is_running {
+            cx.editor
+                .set_status("Debuggee is already running".to_owned());
+            return;
+        }
+
+        let request = debugger.next(debugger.stopped_thread.unwrap());
+        let _ = block_on(request).unwrap();
```

Comment:
Here a crash occurs sometimes. Debugger responds with an error and editor crashes because unwrap gets called on Err. TODO pretty-print errors to status

Reply:
Yeah this `block_on().unwrap()` is temporary, I want to move all of these to `cx.callback(future, || {..})`